arrows/pytorch/iou_tracking.py

```diff
@@ -0,0 +1,63 @@
+class IOU_tracker(object):
+    def __init__(self, iou_accept_threshold, iou_reject_threshold):
+        self._iou_accept_threshold = iou_accept_threshold
+        self._iou_reject_threshold = iou_reject_threshold
+
+    def __call__(self, track_set, track_state_list):
+        return self._track_iou(track_set, track_state_list)
+
+    def _track_iou(self, track_set, track_state_list):
+        # IOU based tracking
+        if track_state_list:
+            for track in track_set.iter_active():
+                # get det with highest iou
+                best_match = max(track_state_list, key=lambda x: self._iou_score(track[-1].bbox, x.bbox))
+
+                # sort the track state list in order to check whether multiple bboxes overlap with the current track
+                sorted_ts_list = sorted(track_state_list, key=lambda x: self._iou_score(track[-1].bbox, x.bbox))
+                best_iou_score = self._iou_score(track[-1].bbox, best_match.bbox)
+
+                if best_iou_score >= self._iou_accept_threshold:
+
+                    # if no other bbox with overlap larger than iou_reject_threshold
+                    if len(sorted_ts_list) >= 2 and self._iou_score(track[-1].bbox, sorted_ts_list[1].bbox) < self._iou_reject_threshold:
```

Comment:
This doesn't seem to test what the comment suggests it should. This line tests that the bbox with the second-*lowest* IOU is below the rejection threshold. Code that performs the test in the comment would be (note the `-2`):
```python
if len(sorted_ts_list) >= 2 and self._iou_score(track[-1].bbox, sorted_ts_list[-2].bbox) < self._iou_reject_threshold:
```
Computing the max *and* sorting is also redundant.